### ISSUE TITLE
EIGRP neighbor between tunnel interfaces

### ISSUE THREAD

**Issue by Tachashi** [batfish/batfish#4090]
When enabling EIGRP on tunnel interfaces in Cisco routers, expected EIGRP neighbor result should be like below:

```
brborder1#sh ip eigrp neighbor
EIGRP-IPv4 Neighbors for AS(1)
H   Address                 Interface              Hold Uptime   SRTT   RTO  Q  Seq
                                                   (sec)         (ms)       Cnt Num
0   10.1.1.1                Tu1                      63 00:28:18   11   100  0  2
```

But in batfish/allinone:latest-79, `bfq.edges()` doesn't see these as EIGRP neighbor:
![bfq edges() result of latest-79](https://user-images.githubusercontent.com/41357228/59546632-1f8bb400-8f6c-11e9-855d-c4aeff080142.png)

At least in early May release, there seemed to have no problem.
![bfq edges() result of early in May](https://user-images.githubusercontent.com/41357228/59546654-c40df600-8f6c-11e9-8dd6-4b05c4d75fa8.png)

[hqborder1_shrun.log](https://github.com/batfish/batfish/files/3292679/hqborder1_shrun.log)
[brborder1_shrun.log](https://github.com/batfish/batfish/files/3292680/brborder1_shrun.log)

**Comment by haverma** [comment]
The reason seems to be that Batfish currently only understands and verifies IPsec tunnels, and not GRE tunnels. The tunnels in the configs use GRE so they don't come up. Earlier it was working because we were activating all Tunnel links (without verifying the Tunnel parameters) but https://github.com/batfish/batfish/pull/3898 removed overlay (Tunnel) links from the topology that `Edges` uses and re-adds only the valid IPsec tunnel links in the post-dataplane topology. We plan to support and verify GRE tunnels as well, very soon.

**Comment by dhalperi** [comment]
@harsh good diagnosis. Is there a way to continue applying the optimistic
reasoning to GRE tunnels until we finish the current work?

**Comment by haverma** [comment]
@dhalperi yes, that should be possible. I can make an exception for the GRE tunnel interfaces while deactivating overlay link interfaces.

**Comment by dhalperi** [comment]
The earlier diagnosis was incomplete:

1. `bfq.eigrpEdges()` returns only edges that only traverse the underlay. This should not ever include edges that cross tunnels; that it did before was a mistake.
   We need to modify this question or add an alternative that reflects the actual established edges.
2. We are not bringing overlay edges between unknown tunnel types (like GRE) back up optimistically. We need to do that, @haverma is working on this.

If we solve both of these, then the given edge should be reflected in the output `bfq.eigrpEdges()` for established edges.

**Comment by yansen14** [comment]
Hi, I am trying to find some references if batfish has supported  for GRE configuration? Recently, I make batfish run and perform about routing tables but find the routes are not learnt if it is coming from GRE tunnel
